oaknational/Oak-Web-Application
# [ENG-343] connect to search api 2023

## Description

- update search hooks to fetch from 2020/2023 search api depending on viewType
- refactors some of the search code (separating: schema -> types -> helpers -> hooks -> components)
- adds search 2023 endpoint to config

## How to test

- run OWA (this branch) [or check preview deploy]
- go to /teachers-2023/search
- perform a search
- should see 2023 search results
- clicking search result should take you to appropriate lesson or unit page
